lib/listing.rb

```diff
@@ -16,9 +16,9 @@ def initialize(listing_id:, name:, description:, price:)
 
   def self.all
 
-    result = database_connection.exec("SELECT * FROM listings;")
+    listing = database_connection.exec("SELECT * FROM listings;")
 
-    result.map do |listing|
+    listing.map do |listing|
```

Comment:
selecting all listings from the table and saving them to the variable listing (singular) and then calling map on that variable where each iteration is also called listing is a bit confusing, and could potentially cause problems. I would save the results of asking the database for everything in the listings table as either result, or listings (plural)

Reply:
Good point! Thank you. I'll make those changes :)